pkg/provisioner/ironic/factory_test.go

```diff
@@ -181,35 +181,78 @@ func TestLoadConfigFromEnv(t *testing.T) {
 	}
 }
 
-func TestLoadEndpointsFromEnv(t *testing.T) {
```

Comment:
Do I understand correctly that you removed this test? Why?
I think we should keep it and add a separate test for the TLS config.